src/qibo/models/circuit.py

```diff
@@ -438,6 +438,31 @@ def decompose(self, *free: int):
             decomp_circuit.add(gate.decompose(*free))
         return decomp_circuit
 
+    def to_clifford(self):
+        """Translate a circuit into an equivalent one composed of only Clifford gates.
+
+        In `Qibo` we refers to [``X``, ``CNOT``, ``RX(pi/2)``, ``RZ(theta)``] as
+        Clifford gates.
+
+        Returns:
+            Circuit object containing only Clifford gates.
+        """
+
+        clifford_queue = []
+        # cycle on gates replacing non-clifford with clifford
+        for gate in self.queue:
+            if gate.is_clifford():
+                clifford_queue.append(gate)
+            else:
+                clifford_queue.extend(gate.decompose_into_clifford())
```

Comment:
Why don't you create the `clifford_circuit` here and add the gates directly, instead of creating a list and then adding it to the circuit?
```suggestion
        clifford_circuit = self.__class__(**self.init_kwargs)
        # cycle on gates replacing non-clifford with clifford
        for gate in self.queue:
            if gate.clifford:
                clifford_circuit.add(gate)
            else:
                clifford_circuit.add(gate.decompose_into_clifford())
```
Note that `circuit.add()` supports lists or iterables of gates (it will automatically loop over the iterable and add all gates).